src/scripts/extensions/extensionBase.ts

```
@@ -240,14 +240,14 @@ export abstract class ExtensionBase<TWorker extends ExtensionWorkerBase<TTab, TT
 		this.updateClientInfoWithFlightInformation([]);
 	}
 
-	private shouldShowTooltip(tab: TTab, tooltipTypes: TooltipType[]): TooltipType {
+	private async shouldShowTooltip(tab: TTab, tooltipTypes: TooltipType[]): Promise<TooltipType> {
 		let type = this.checkIfTabMatchesATooltipType(tab, tooltipTypes);
 
 		if (!type) {
```

should we return a failed promise instead of plain return?